frontend/apps/web/app/(mgmt)/[account]/new/job/generate/single/connect/page.tsx

```diff
@@ -145,71 +161,117 @@ export default function Page({ searchParams }: PageProps): ReactElement {
+                                const destId = form.getValues(
+                                  'destination.connectionId'
+                                );
+
+                                const urlParams = new URLSearchParams({
+                                  returnTo: `/${account?.name}/new/job/generate/single/connect?sessionId=${sessionPrefix}&from=new-connection`,
+                                });
+                                const connection = connections.find(
+                                  (c) => c.id === destId
+                                );
+                                const connType = getConnectionType(
+                                  connection?.connectionConfig ??
+                                    new ConnectionConfig()
+                                );
+
+                                if (connType) {
+                                  urlParams.append('connectionType', connType);
+                                }
+                                if (
+                                  urlParams.getAll('connectionType').length ===
+                                  0
+                                ) {
+                                  urlParams.append(
+                                    'connectionType',
+                                    'pgConfig'
+                                  );
+                                  urlParams.append(
+                                    'connectionType',
+                                    'mysqlConfig'
+                                  );
+                                }
+                                router.push(
+                                  `/${account?.name}/new/connection?${urlParams.toString()}`
+                                );
+                                return;
+                              }
+                              field.onChange(value);
                               const destId = form.getValues(
                                 'destination.connectionId'
                               );
 
-                              const urlParams = new URLSearchParams({
-                                returnTo: `/${account?.name}/new/job/generate/single/connect?sessionId=${sessionPrefix}&from=new-connection`,
-                              });
-                              const connection = connections.find(
-                                (c) => c.id === destId
-                              );
-                              const connType = getConnectionType(
-                                connection?.connectionConfig ??
-                                  new ConnectionConfig()
-                              );
-                              if (connType) {
-                                urlParams.append('connectionType', connType);
+                              setIsSourceValidating(true);
+                              try {
+                                const res = await checkConnectionConfig({
```

Comment:
This is going to block the UI from updating the `destination.connectionId` until this returns, which could be bad if the connection is broken or some other reason.

In other words, this should be not awaited here and instead done as a standard promise, in a separate effect, or the below code should be moved to above this.